mlflow/utils/process.py

```diff
@@ -75,6 +77,18 @@ def _exec_cmd(
 
     env = env if extra_env is None else {**os.environ, **extra_env}
 
+    if is_in_databricks_runtime():
```

Comment:
1. Can we add a comment for what this is for? Alternatively, can extract to self-explanatory function like `_remove_inaccesible_python_path`.
2. Why this needs to be databricks untime only?

Reply:
@WeichenXu123 Sorry for the back-and-forth, but we may need to limit it to in Databricks on second thought.... Here we mutate the environment variable in-place, which is probably fine in Databricks where most users should not care `PYTHONPATH` (mostly in our control). However, in OSS ppl may have their own environment set up and we should not break that.